### PR TITLE
API-907: Fix Apps audit API response format

### PR DESCRIPTION
**Pull request by pchasle**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Apps/back/Domain/Audit/Model/Read/DailyEventCount.php [hunk 1]
```diff
@@ -17,7 +17,7 @@ final class DailyEventCount
     /** @var \Datetime */
     private $date;
 
-    public function __construct(int $count, \DateTime $date)
+    public function __construct(int $count, \DateTimeInterface $date)
```
Comment (fitn) [inline src/Akeneo/Apps/back/Domain/Audit/Model/Read/DailyEventCount.php]: That's my fault but do we need \Datetime here? Maybe a string would be enough?

File: src/Akeneo/Apps/back/Infrastructure/Persistence/Dbal/Query/DbalSelectAppsEventCountByDayQuery.php [hunk 2]
```diff
@@ -31,37 +31,94 @@ public function execute(string $eventType, string $startDate, string $endDate):
...
-            'end_date'   => $endDateTime->format('Y-m-d'),
-            'event_type' => $eventType
+            'end_date' => $endDateTime->format('Y-m-d'),
+            'event_type' => $eventType,
         ];
 
-        $dataRows = $this->dbalConnection->executeQuery($sqlQuery, $sqlParams)->fetchAll();
+        $result = $this->dbalConnection->executeQuery($sqlQuery, $sqlParams)->fetchAll();
+
+        $eventCountsDataPerApp = $this->fillMissingDates(
```
Comment (fitn) [inline src/Akeneo/Apps/back/Infrastructure/Persistence/Dbal/Query/DbalSelectAppsEventCountByDayQuery.php]: I fill this should be extracted in a specific hydration object/class. That will be far more easy to test :)

File: src/Akeneo/Apps/back/Domain/Audit/Model/Read/DailyEventCount.php (hunk 1 above)
Comment (pchasle) [reply src/Akeneo/Apps/back/Domain/Audit/Model/Read/DailyEventCount.php]: It's fine :)

File: src/Akeneo/Apps/back/Infrastructure/Persistence/Dbal/Query/DbalSelectAppsEventCountByDayQuery.php (hunk 2 above)
Comment (pchasle) [reply src/Akeneo/Apps/back/Infrastructure/Persistence/Dbal/Query/DbalSelectAppsEventCountByDayQuery.php]: Following our discussion, should I rewrite this now?

**Pull request merged**